src/supertux/flip_level_transformer.cpp

```diff
@@ -30,9 +31,19 @@ void
 FlipLevelTransformer::transform_sector(Sector& sector)
 {
   float height = sector.get_height();
+  std::set<Path*> paths;
 
   for (auto& object : sector.get_objects()) {
     object->on_flip(height);
+    if (Path* path = object->get_path_if_exists()) {
```

Comment:
All objects that have a path are `PathObject`'s; you can use a `dynamic_cast<PathObject*>(object)` here to check if the object is a PathObject and use it afterwards.

(I think there's one or two objects that have paths but don't inherit from PathObject, but don't take them in consideration, these objects are the ones that need to be fixed)

Reply:
Thank you! Those are the C++ things I will have to learn. I knew there must be a nicer solution.